### ISSUE TITLE
[BUG] Field “thumbnail” must not have a selection since type “String” has no subfields

### ISSUE THREAD

**Issue by leochoo** [gatsbyjs/gatsby#31990]
TL DR: 
1. This bug is NOT solved. This bug makes gatsby code inconsistent and unreliable across environments.
2. Even the official `gatsby-starter-blog` fails.
3. It requires a serious attention from gatsby.

## Description

This bug is very frustrating because this error happens often but not always.
This bug makes gatsby code inconsistent and unreliable across environments.
It can happen on the same machine. Exact same production code, one day it works, one day it does not and I have to do `gatsby clean` to fix the issue. But then, even this does not work on some environments. I wasted so much time because of this.

Even the [gatsby-starter-blog example shows this error](https://github.com/gatsbyjs/gatsby/issues/4123#issuecomment-771925663). I faced the exact same issue as well.

I realized that this issue was closed, but considering how this problem persists, the issues needs to be re-opened and requires attention from gatsby to be fixed.

Please address this as soon as possible.

I have been getting this error in multiple devices and environments.

Exact same production code, `git pull`  

What's frustrating is that the exact same production code will build on some machines directly after cloning,
but on some machine it won't.

It usually gets resolved after `gatsby clean` on Mac OS. However on Windows, this still does NOT work.

This issue is clearly not solved, but it is marked as closed in below ticket.
https://github.com/gatsbyjs/gatsby/issues/4123

### Steps to reproduce

https://github.com/gatsbyjs/gatsby/issues/4123#issuecomment-771925663

### Expected result

It should build

### Actual result

This error.

**Comment by LekoArts** [comment]
Hi!

Please create a [minimal reproduction][reproduction] showing your problem. The example you linked is incorrect (I commented on it).

[reproduction]: https://gatsby.dev/reproduction

**Comment by LekoArts** [comment]
Hi!

Since we didn't receive an answer for 7 days or more about the ask of a reproduction I'm going to close this now, as we can't do much to help without a reproduction. If you are able to [create a minimal reproduction](https://gatsby.dev/reproduction) for this then please do answer here or open a new issue with a reproduction. Thanks!
